### ISSUE TITLE
Translation For Terms/Definitions

### ISSUE THREAD

**Issue by grimsteel** [For-0/vocabustudy#150]
Automatically translate term to definition (and also set language for term/definition)

**Comment by grimsteel** [comment]
I was doing some procrastination and came across this:
https://www.tensorflow.org/text/tutorials/nmt_with_attention
I've learnt about TF in theory (I've read many books) but I've never used it in practice
We would need to make a model for English - Spanish rather than Spanish - English and we would have to port it to TF.js
@IMGROOT2 @DG3QRT1 @Nikhilatvocabustudy @ShreyanKhanna Does anybody else here have better experience with TF?

**Comment by IMGROOT2** [comment]
I have no clue, lol. But can I be rlly honest with you? I feel like translation should be pushed to 2.0.0 or future- its a feature will not be used that much. anyway, we want the flashcards to be english AND Spanish for the Spanish sets, no?

i think it's more complicated than we think, and we should spend more time enhancing more important issues. also, are we deploying stable anytime soon? i don't think 0.6.5 has any bugs that we know of, so can we just launch 0.6.1 - 0.6.5?

**Comment by IMGROOT2** [comment]
i think none of our users want to actually change the language from english to spanish. are we just dealing with english speakers?

**Comment by grimsteel** [comment]
I think this feature was more in response to the quizlet feature in the set editor where you can type a term and it will auto suggest the translated definition

But overall I feel the same - I have different priorities

**Comment by IMGROOT2** [comment]
it's definitely a useful tool, but it's just not the right time yet

**Comment by grimsteel** [comment]
As for translating the _entire site_ into different languages, there's a tool for that, but it costs $100+ per month...
Luckily we have it for free with the student developer pack!
![image](https://user-images.githubusercontent.com/115331356/200732042-2af1bf32-da70-484f-ac3a-d9df1a229ced.png)

**Comment by ShreyanKhanna** [comment]
This is cool… idk how u came across this but maybe in the far future bc if u look at other sites with this, they started using it as their platform grew super large and idk if the vocabustudy platform is that large rn

**Comment by ShreyanKhanna** [comment]
> > it's definitely a useful tool, but it's just not the right time yet
> 
> Yes, pretty much all of the spanish sets are being copied from Profe Cuevas's quizlet account lol

Yea but isn’t vocabustudy a free copy of quizlet (that’s better) anyways

**Comment by ShreyanKhanna** [comment]
Also I’m sorry for my inactivity, I really want to start help y’all but other things like me being sick and getting contacts got in the way, but fortunately I’m ready to help whenever y’all need it. I just don’t know what to do to help.. can any of y’all help tell me what I can do or where our plans are for the future. Thanks :)

**Comment by grimsteel** [comment]
I understand - I've been feeling REALLY tired over the past few weeks, and all the US History HW isn't helping. You can just do whatever you can - #1 is a pretty easy one - you just have to make sure everything looks good on phones and point out things that don't

As for the future, this is a pretty good summary: https://github.com/orgs/For-0/projects/2/views/3

**Comment by IMGROOT2** [comment]
i HATE US HISTORY!!! 

just needed to get that out.


my entire family is asleep.

i think I'm going insane...

anyway!

it's perfectly fine @ShreyanKhanna, lol. i agree with @grimsteel; #1 is a simple one (I should apologize for not doing that already). just fire up webstorm or whatever, and publish to [this branch.](https://github.com/For-0/vocabustudy/tree/1-responsive-design-on-all-pages) I'll do the same, too.

**Comment by IMGROOT2** [comment]
![image](https://user-images.githubusercontent.com/116324098/200747765-6a6e6854-5219-497a-a405-2399a5b6952b.jpeg)

**Comment by ShreyanKhanna** [comment]
Wait is Siddhant the bot?

**Comment by ShreyanKhanna** [comment]
> I understand - I've been feeling REALLY tired over the past few weeks, and all the US History HW isn't helping. You can just do whatever you can - #1 is a pretty easy one - you just have to make sure everything looks good on phones and point out things that don't
> 
> As for the future, this is a pretty good summary: https://github.com/orgs/For-0/projects/2/views/3

Alright I’ll get started as soon as I can

**Comment by grimsteel** [comment]
> Wait is Siddhant the bot?

Yes

**Comment by ShreyanKhanna** [comment]
Bro did u take one online or code ur own?

**Comment by grimsteel** [comment]
My own ofc...
It's a super simple bot anyway

**Comment by ShreyanKhanna** [comment]
Makes sense cuz it would personally be faster to code ur own than to find one that works online

**Comment by IMGROOT2** [comment]
for my bot i used autocode (shame on me, ik), but that's because I only need to update the rules, which should ideally never need to happen.

but it took literally 2 minutes to find a bot online, so there was no reason for me to code it and everything. in siddhant's case, he needs to send regular and frequent reminders

**Comment by IMGROOT2** [comment]
ok i just installed jetbrains gateway, which is essentially the local-end connection to github codespaces and other remote-dev platforms. whenever I get codespaces, I'll get it set up immediately.

super excited as you can tell
